includes/reader-revenue/class-stripe-sync.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Stripe Syncs.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Stripe Syncs.
+ */
+class Stripe_Sync {
+	/**
+	 * The final results object.
+	 *
+	 * @var array
+	 * @codeCoverageIgnore
+	 */
+	private static $results = [
+		'processed'         => 0,
+		'created'           => 0,
+		'found'             => 0,
+		'orders_updated'    => 0,
+		'orders_created'    => 0,
+		'skipped_customers' => 0,
+	];
+
+	/**
+	 * Initialize.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public static function init() {
+		\add_action( 'init', [ __CLASS__, 'wp_cli' ] );
+	}
+
+	/**
+	 * Process a Stripe customer.
+	 *
+	 * @param array $customer Customer object.
+	 * @param array $args Arguments.
+	 */
+	private static function process_stripe_customer( $customer, $args ) {
+		$email_address = $customer->email;
+		$is_dry_run    = false !== $args['dry-run'];
+
+		$all_charges = Stripe_Connection::get_customer_charges( $customer->id );
+
+		// Skip charges created by WC.
+		$charges = array_filter(
+			$all_charges,
+			function( $charge ) {
+				return ! isset( $charge->metadata->order_id );
+			}
+		);
+		if ( empty( $charges ) ) {
+			self::$results['skipped_customers']++;
+			return;
+		}
+
+		$wp_user = get_user_by( 'email', $email_address );
+		$user_id = false;
+		if ( $wp_user ) {
+			$user_id = $wp_user->ID;
+			self::$results['found']++;
+		} else {
+			// Create the WC Customer and update past orders (lookup is done by email).
+			if ( ! $is_dry_run ) {
+				$full_name  = $customer->name;
+				$user_login = \sanitize_title( $full_name );
+				$user_id    = \wc_create_new_customer( $email_address, $user_login, '', [ 'display_name' => $full_name ] );
+				if ( is_wp_error( $user_id ) ) {
+					\WP_CLI::warning( __( 'Error processing customer', 'newspack' ) . ' ' . $email_address . ': ' . $user_id->get_error_message() );
```

Comment:
```
Warning: Error processing customer [redacted]@mail.com: An account is already registered with that username. Please choose another.
```
There is a somewhat rare situation I ran into where if there are two people with the same name (e.g. two people named Joe Smith), this throws an error. I think a `get_user_by( 'login', $user_login )` check here could avoid that situation. I'm not sure the best fallback username though (maybe something email based or add some suffix to the username).

Reply:
Thanks for catching that! Amended in 2f32891ccd21d4c72c0396190409fadefb22db2c so that an existing username will get a unique-id suffix.